compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -443,7 +444,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         };
         let def_id = method_item.def_id;
         let generics = tcx.generics_of(def_id);
-        assert_eq!(generics.params.len(), 0);
+
+        if generics.params.len() != 0 {
+            tcx.sess.emit_fatal(OpMethodGenericParams {
```

Comment:
Does this need to be fatal? Fatal errors should be reserved for particularly unrecoverable positions, and this one doesn't seem (at least initially) like one to me.

We could just return `None` here -- what does that do to the error reporting?

Reply:
With a basic `emit_err` it causes further ICEs in other places. I haven't tried returning `None` but I assume it's gonna emit further errors about things not being found.

This code path is only hit when using `no_core` in weird ways (like by being a fuzzer) so I don't think we should care too much about recovery.